groups/bsl/bslstl/bslstl_badvariantaccess.h

```diff
@@ -0,0 +1,103 @@
+// bslstl_badvariantaccess.h                                          -*-C++-*-
+#ifndef INCLUDED_BSLSTL_BADVARIANTACCESS
+#define INCLUDED_BSLSTL_BADVARIANTACCESS
+
+#include <bsls_ident.h>
+BSLS_IDENT("$Id: $")
+
+//@PURPOSE: Provide an exception class thrown by 'bsl::variant'.
+//
+//@CLASSES:
+//  bsl::bad_variant_access: exception type thrown by 'bsl::variant'
+//
+//@SEE_ALSO: bslstl_variant, bslstl_stdexceptionutil
+//
+//@DESCRIPTION: This component provides a 'bsl::bad_variant_access' exception
+// class.  This exception is thrown by 'bsl::variant::value' when accessing a
+// 'bsl::variant' object that does not contain a value.  If 'std::variant'
+// implementation is available, 'bsl::bad_variant_access' is an alias to
+// 'std::bad_variant_access'.
+//
+
+#include <bslscm_version.h>
+
+#include <bsls_exceptionutil.h>
+#include <bsls_keyword.h>
+#include <bsls_libraryfeatures.h>
+
+#include <exception>
+
+#ifdef BDE_BUILD_TARGET_EXC
+
+#ifdef BSLS_LIBRARYFEATURES_HAS_CPP17_BASELINE_LIBRARY
+#include <variant>  // for 'std::bad_variant_access'
+#endif              // BSLS_LIBRARYFEATURES_HAS_CPP17_BASELINE_LIBRARY
+
+namespace bsl {
+#ifdef BSLS_LIBRARYFEATURES_HAS_CPP17_BASELINE_LIBRARY
+typedef std::bad_variant_access bad_variant_access;
+#else
+                          // ========================
+                          // class bad_variant_access
+                          // ========================
+
+class bad_variant_access : public std::exception {
+  public:
+    // CREATORS
+    bad_variant_access() BSLS_KEYWORD_NOEXCEPT;
+        // Create a 'bad_variant_access' object.  Note that this function is
+        // explicitly user-declared, to make it simple to declare 'const'
+        // objects of this type.
+
+    // ACCESSORS
+    const char *what() const BSLS_EXCEPTION_WHAT_NOTHROW BSLS_KEYWORD_OVERRIDE;
```

Comment:
See https://github.com/bloomberg/bde/blob/main/groups/bsl/bsls/bsls_exceptionutil.h#L317
```suggestion
    const char *what() const BSLS_EXCEPTION_VIRTUAL_NOTHROW
                                                         BSLS_KEYWORD_OVERRIDE;
```